starfish/spots/_detector/local_search_blob_detector.py

```diff
@@ -0,0 +1,423 @@
+            round_index = round_index[mask]
+            intensity_data = intensity_data[mask]
+
+            # need numpy indexing to set values in vectorized manner
+            intensity_table.values[feature_index, channel_index, round_index] = intensity_data
+
+        return intensity_table
+
+    def run(
+        self, data: ImageStack, verbose: bool=False, n_processes: Optional[int]=None
+    ) -> IntensityTable:
+        """Find 1-hot coded spots in data.
+
+        Parameters
+        ----------
+        data : ImageStack
+            Image data containing coded spots.
+        verbose : bool
+            If True, report on progress of spot finding.
+        n_processes : Optional[int]
+            Number of processes to devote to spot finding. If None, will use the number of available
+            cpus (Default None).
+
+        Returns
+        -------
+        IntensityTable
+            Contains detected coded spots.
+
+        """
+        per_tile_spot_results = self._find_spots(data, verbose=verbose, n_processes=n_processes)
+
+        per_round_spot_results = self._merge_spots_by_round(per_tile_spot_results)
+
+        distances, indices = self._match_spots(
+            per_round_spot_results,
+            search_radius=self.search_radius, anchor_round=self.anchor_round
+        )
+
+        # TODO implement consensus seeding (SeqFISH)
+
+        intensity_table = self._build_intensity_table(
+            per_round_spot_results, distances, indices,
+            rounds=data.xarray[Axes.ROUND.value].values, channels=data.xarray[Axes.CH.value].values,
+            search_radius=self.search_radius, anchor_round=self.anchor_round
+        )
+
+        transfer_physical_coords_from_imagestack_to_intensity_table(
+            image_stack=data, intensity_table=intensity_table
+        )
+
+        return intensity_table
+
+    @staticmethod
+    @click.command("LocalSearchBlobDetector")
+    @click.option(
+        "--min-sigma", default=4, type=int, help="Minimum spot size (in standard deviation).")
+    @click.option(
+        "--max-sigma", default=6, type=int, help="Maximum spot size (in standard deviation).")
+    @click.option(
+        "--threshold", default=.01, type=float, help="Dots threshold.")
+    @click.option(
+        "--overlap", default=0.5, type=float,
+        help="Dots with overlap of greater than this fraction are combined.")
+    @click.option(
+        "--detector-method", default='blob_log', type=Choice(['blob_log', 'blob_dog']),
+        help="Name of the type of the skimage blob detection method.")
+    @click.option(
+        "--search-radius", default=3, type=int,
+        help="Number of pixels over which to search for spots in other image tiles.")
+    @click.pass_context
+    def _cli(
+        ctx, min_sigma, max_sigma, threshold, overlap, show, detector_method, search_radius
+    ) -> None:
+        instance = LocalSearchBlobDetector(
+            min_sigma, max_sigma, threshold, overlap,
```

Comment:
What's the story with sigma?  I see sometimes that it's allowed to be a tuple and sometimes it has to be a scalar.  In any case, this cli setup does not allow this to be a tuple.